PaddleCV/PaddleDetection/ppdet/experimental/mixed_precision.py

```diff
@@ -0,0 +1,324 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+import six
+from paddle.fluid.framework import Parameter
+from paddle.fluid import layers
+from paddle.fluid import core
+from paddle.fluid import unique_name
+import paddle.fluid.layer_helper_base as lhb
+import paddle.fluid.optimizer as optim
+
+__all__ = ['mixed_precision_global_state', 'mixed_precision_context',
+           'StaticLossScale', 'DynamicLossScale']
+
+_mixed_precision_global_state = None
+
+
+def mixed_precision_global_state():
+    return _mixed_precision_global_state
+
+
+class LossScale(object):
+    def __init__(self):
+        super(LossScale, self).__init__()
+
+    def get_loss_scale_var(self):
+        return self.scale
+
+    def increment(self):
+        raise NotImplementedError()
+
+    def decrement(self):
+        raise NotImplementedError()
+
+
+class StaticLossScale(LossScale):
+    """
+    Static (fixed) loss scale manager.
+
+    Args:
+        init_loss_scale (float): initial loss scale value.
+
+    Examples:
+
+        .. code-block:: python
+
+            from paddle import fluid
+            from paddle.fluid.mixed_precision import (mixed_precision_context,
+                                                      StaticLossScale)
+
+            with mixed_precision_context(StaticLossScale(8.), True) as ctx:
+                # ...
+                # scale loss
+                loss_scale = ctx.get_loss_scale_var()
+
+    """
+
+    def __init__(self, init_loss_scale=1.):
+        super(StaticLossScale, self).__init__()
+        self.scale = layers.create_global_var(
+            name=unique_name.generate("loss_scale"),
+            shape=[1],
+            value=init_loss_scale,
+            dtype='float32',
+            persistable=True)
+
+
+class DynamicLossScale(LossScale):
+    """
+    Dynamic loss scale manager. it works as follows:
+    if gradients is valid for `increment_every` steps, loss scale values is
+    increased by `factor`, otherwise loss scale values is decreased by `factor`
+
+    Args:
+        init_loss_scale (float): initial loss scale value.
+        increment_every (int): minimum 'good' steps before increase loss scale.
+        factor (float): increase/decrease loss scale by this much.
+
+    Examples:
+
+        .. code-block:: python
+
+            from paddle import fluid
+            from paddle.fluid.mixed_precision import (mixed_precision_context,
+                                                      StaticLossScale)
+
+            loss_scale = DynamicLossScale(8., 1000, 4.)
+            with mixed_precision_context(loss_scale, True) as ctx:
+                # ...
+                # scale loss
+                loss_scale = ctx.get_loss_scale_var()
+
+    """
+
+    def __init__(self, init_loss_scale=2**15, increment_every=2000, factor=2.):
```

Comment:
increment_every -> increment_every_step ？

Reply:
could be either this or `increment_every_n_steps`,  the former is shorter and the meaning should be quite clear.